# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/itz-me-sree/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [32]:
!git clone https://github.com/itz-me-sree/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (75/75), done.
remote: Total 119 (delta 35), reused 95 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (119/119), 1.84 MiB | 6.74 MiB/s, done.
Resolving deltas: 100% (35/35), done.


In [33]:
%cd flyrank-ml-internship


/content/flyrank-ml-internship/flyrank-ml-internship/flyrank-ml-internship


In [34]:
!find . -name "content_refresh_anonymized.csv"

./data/raw/content_refresh_anonymized.csv


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [35]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.



## My Lane

I chose the SEO Content Optimization lane.

I am interested in understanding which pages are likely to lose performance so that they can be improved before traffic declines. This can help prioritize content updates instead of reviewing every page manually.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [36]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Decision:
Which pages should be reviewed first?

Action:
Update or optimize pages that are likely to decline.

Cost of a wrong recommendation:
If a healthy page is selected, time is wasted.
If a declining page is ignored, search traffic and clicks may decrease.

Why ML can help:
Large websites contain thousands of pages, making manual review difficult. ML can help identify patterns and prioritize pages that need attention.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [37]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average Search Volume:", round(df["search_volume"].mean(), 2))

Rows: 30000
Columns: 44
Average CTR: 0.511
Average Search Volume: 158.88


The starter dataset contains 30,000 pages and 44 features.

The average CTR is 0.511 and the average search volume is 158.88.

These numbers suggest that page performance varies across the dataset, making it useful for prioritizing which pages should be reviewed first.

The starter dataset contains thousands of pages and multiple SEO-related features.

The average CTR and search volume show that page performance varies significantly, making this a suitable problem for prioritization.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [38]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

This notebook only explores patterns in the available data.

It does not prove how Google's ranking algorithm works.

The observations are based only on the provided dataset.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.